src/api/recordlinker/main.py

```diff
@@ -0,0 +1,95 @@
+api = fastapi.FastAPI(
+    title="Record Linker API",
+    version=__version__,
+    contact={
+        "name": "CDC Public Health Data Infrastructure",
+        "url": "https://github.com/CDCgov/RecordLinker",
+    },
+    license_info={
+        "name": "Apache 2.0",
+        "url": "https://www.apache.org/licenses/LICENSE-2.0",
+    },
+    summary="""
+        The RecordLinker is a service that links records from two datasets based on a set
+        of common attributes. The service is designed to be used in a variety of public
+        health contexts, such as linking patient records from different sources or linking
+        records from different public health surveillance systems. The service uses a
+        probabilistic record linkage algorithm to determine the likelihood that two
+        records refer to the same entity. The service is implemented as a RESTful API that
+        can be accessed over HTTP. The API provides endpoints for uploading datasets,
+        configuring the record linkage process, and retrieving the results of the record
+        linkage process.
+    """.strip(),
+)
+
+api.add_middleware(middleware.CorrelationIdMiddleware)
+api.add_middleware(middleware.AccessLogMiddleware)
+if settings.ui_host:
+    # Add CORS for local development
+    api.add_middleware(
+        CORSMiddleware,
+        allow_origins=[settings.ui_host],
+        allow_credentials=True,
+        allow_methods=["*"],
+        allow_headers=["*"],
+    )
+
+
+# FIXME: Change health check endpoint to /api/health
+api.include_router(health_router)
+api.include_router(link_router, tags=["link"])
+api.include_router(algorithm_router, prefix="/algorithm", tags=["algorithm"])
+api.include_router(person_router, prefix="/person", tags=["mpi"])
+api.include_router(patient_router, prefix="/patient", tags=["mpi"])
+api.include_router(seed_router, prefix="/seed", tags=["mpi"])
+
+# FIXME: This is going to break the NBS integration, we need to communicate this
+# well in advance
+app.mount("/api", api)
+
+if settings.ui_static_dir:
+    # static files for the UI
+    app.mount(
+        "/_next",
+        StaticFiles(directory=os.path.join(settings.ui_static_dir, "_next")),
+        name="SpaStaticAssets",
+    )
+
+    @app.exception_handler(StarletteHTTPException)
+    async def not_found_handler(request, exc):
+        """
+        Custom 404 page
+        """
+        if exc.status_code == fastapi.status.HTTP_404_NOT_FOUND:
+            return FileResponse(os.path.join(settings.ui_static_dir, "404.html"), status_code=404)
+        raise exc
+
+    @app.get("/")
+    @app.get("/wizard")
+    @app.get("/favicon.ico")
+    async def page(request: fastapi.Request):
+        """
+        Route to handle custom HTML pages for the UI
+        """
+        path: str = request.url.path.strip("/") or "index"
+        view: str = path if path.endswith(".ico") else f"{path}.html"
+        return FileResponse(os.path.join(settings.ui_static_dir, view))
```

Comment:
This is actually really neat. I was wondering if a dynamic setup was possible.

Reply:
I went back and forth on how dynamic to make this.  One option I considered was to just allow for a path variable (eg `@app.get("/{page}")`), parse that value and determine the appropriate HTML page to generate.  The advantage there being as more pages are created, we don't need to update this method.  My instinct though was this could lead to a security vulnerability, and it might just be best to list all the **known** pages as explicit routes (listed 86-88).  I mean, we're likely to only have a half dozen or so for this project, so that seems fine.